service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/IdMapper.java

```diff
@@ -55,19 +70,26 @@ public String get(Reference reference) {
             var entry = replacement.get();
             LOGGER.warn("Replacing unmapped resource {} with {} => {}", referenceValue,
                 entry.getKey(), entry.getValue());
-            return entry.getValue();
+            return entry.getValue().getId();
         }
 
         LOGGER.warn("Unable to find a replacement resource for {}", referenceValue);
         return null;
     }
 
-    public String get(ResourceType resourceType, String id) throws EhrMapperException {
+    public String get(ResourceType resourceType, IdType id) throws EhrMapperException {
         return get(buildReference(resourceType, id));
     }
 
-    public static Reference buildReference(ResourceType resourceType, String id) {
-        return new Reference(new IdType(resourceType.name(), id));
+    private static Reference buildReference(ResourceType resourceType, IdType id) {
+        return new Reference(new IdType(resourceType.name(), id.getIdPart()));
     }
 
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    private static class MappedId {
+        private String id;
+        private Boolean isResourceMapped;
```

Comment:
Minor memory and performance optimisation if you use the primitive
```suggestion
        private boolean isResourceMapped;
```
ref: https://www.baeldung.com/java-primitives-vs-objects